components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +132,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.where(column, OperatorType.LIKE, "%");
```

Comment:
As far as I know (I might be wrong) then '%' will also match the empty string, which would make this wrong?